homeassistant/components/advantage_air/switch.py

```diff
@@ -23,10 +23,11 @@ async def async_setup_entry(
 
     instance = hass.data[ADVANTAGE_AIR_DOMAIN][config_entry.entry_id]
 
-    entities = []
-    for ac_key, ac_device in instance["coordinator"].data["aircons"].items():
-        if ac_device["info"]["freshAirStatus"] != "none":
-            entities.append(AdvantageAirFreshAir(instance, ac_key))
+    entities: list[SwitchEntity] = []
+    if "aircons" in instance["coordinator"].data:
```

Comment:
This applies in multiple locations: maybe a walrus would be prettier here, which would make the next line easier to digest:
```suggestion
    if aircons := instance["coordinator"].data.get("aircons"):
```

Reply:
Good idea, I've adopted this throughout.